includes/data-port/export-tasks/class-sensei-export-lessons.php

```diff
@@ -0,0 +1,131 @@
+		return 'lesson';
+	}
+
+	/**
+	 * Collect exported fields of the lesson.
+	 *
+	 * @param WP_Post $post The lesson.
+	 *
+	 * @return string[]
+	 */
+	protected function get_post_fields( $post ) {
+		$meta           = [];
+		$meta_keys      = [
+			'_lesson_course',
+			'_lesson_prerequisite',
+			'_lesson_preview',
+			'_lesson_length',
+			'_lesson_complexity',
+			'_lesson_video_embed',
+		];
+		$quiz_meta_keys = [
+			'_pass_required',
+			'_quiz_passmark',
+			'_show_questions',
+			'_random_question_order',
+			'_quiz_grade_type',
+			'_enable_quiz_reset',
+		];
+
+		foreach ( $meta_keys as $meta_key ) {
+			$meta[ $meta_key ] = get_post_meta( $post->ID, $meta_key, true );
+		}
+
+		$quiz_id = Sensei()->lesson->lesson_quizzes( $post->ID );
+
+		foreach ( $quiz_meta_keys as $meta_key ) {
+			$meta[ $meta_key ] = get_post_meta( $quiz_id, $meta_key, true );
+		}
+
+		$tags           = get_the_terms( $post->ID, 'lesson-tag' );
+		$module_term_id = Sensei()->modules->get_lesson_module_if_exists( $post );
+		$questions_ids  = $this->get_quiz_question_ids( $quiz_id );
+
+		$columns = [
+			Schema::COLUMN_ID             => $post->ID,
+			Schema::COLUMN_TITLE          => $post->post_title,
+			Schema::COLUMN_SLUG           => $post->post_name,
+			Schema::COLUMN_DESCRIPTION    => $post->post_content,
+			Schema::COLUMN_EXCERPT        => $post->post_excerpt,
+			Schema::COLUMN_STATUS         => $post->post_status,
+			Schema::COLUMN_COURSE         => Sensei_Data_Port_Utilities::serialize_id_field( $meta['_lesson_course'] ),
+			Schema::COLUMN_MODULE         => 0 !== $module_term_id ? get_term( $module_term_id )->name : '',
+			Schema::COLUMN_PREREQUISITE   => Sensei_Data_Port_Utilities::serialize_id_field( $meta['_lesson_prerequisite'] ),
+			Schema::COLUMN_PREVIEW        => 'preview' === $meta['_lesson_preview'] ? 1 : 0,
+			Schema::COLUMN_TAGS           => ! empty( $tags ) ? Sensei_Data_Port_Utilities::serialize_term_list( $tags ) : '',
+			Schema::COLUMN_IMAGE          => get_the_post_thumbnail_url( $post, 'full' ),
+			Schema::COLUMN_LENGTH         => $meta['_lesson_length'],
+			Schema::COLUMN_COMPLEXITY     => $meta['_lesson_complexity'],
+			Schema::COLUMN_VIDEO          => $meta['_lesson_video_embed'],
+			Schema::COLUMN_PASS_REQUIRED  => 'on' === $meta['_pass_required'] ? 1 : 0,
+			Schema::COLUMN_PASSMARK       => $meta['_quiz_passmark'],
+			Schema::COLUMN_NUM_QUESTIONS  => $meta['_show_questions'],
+			Schema::COLUMN_RANDOMIZE      => 'yes' === $meta['_random_question_order'] ? 1 : 0,
+			Schema::COLUMN_AUTO_GRADE     => 'manual' === $meta['_quiz_grade_type'] ? 0 : 1,
+			Schema::COLUMN_QUIZ_RESET     => 'on' === $meta['_enable_quiz_reset'] ? 1 : 0,
+			Schema::COLUMN_ALLOW_COMMENTS => 'closed' === $post->comment_status ? 0 : 1,
+			Schema::COLUMN_QUESTIONS      => Sensei_Data_Port_Utilities::serialize_id_field( $questions_ids ),
+		];
+
+		$schema = array_keys( $this->get_type_schema()->get_schema() );
+		return array_map(
```

Comment:
I quite like this, but can see an argument for doing it up in the parent class when `get_post_fields` is called. The return for the courses export seems pretty brittle to me. We'll one day want to allow for fields/schema to be filterable and plugins to add fields (like WCPC to be able to add product ID). This would be easier if we return a column key'd array here and then had the parent class build the row based on the schema.

Reply:
Good idea! Refactored here: https://github.com/Automattic/sensei/pull/3513/commits/ae813721182d75599605a04c024bc77e2fd3d1e4